FuelLabs/fuel-indexer
# Allow for optional ABI JSON tokens in #[indexer] codegen

Closes #289.

## Changelog
- Enable optional ABI JSON tokens in #[indexer] codegen
- Remove default ABI JSON file
- Add additional logging to WASM executor
- Fix path in block explorer example schema

## Testing Plan
_No ABI path (using block explorer example)_
1. `cargo build -p fuel-indexer-test-web --release`
2. Build block explorer WASM or use build test WASM script
3. `bash scripts/utils/start_test_components.bash`
4. `cargo run --bin fuel-indexer -- --manifest examples/block-explorer/explorer_index.manifest.yaml`
5. `bash examples/block-explorer/go.bash` to generate blocks
6. Watch the blocks get picked up by the indexer.
7. Check database to ensure that they are present.

_With ABI path_
- Check other examples as one would normally do

## Notes
I spent quite awhile trying to figure out why the WASM module wasn't working, and then I realized that we were panicking on any value that didn't match the types in the ABI. Thus, when there is no content in the ABI, everything is caught by the `_` match arm. To prevent this (and to create a slightly more ergonomic experience), I've removed the `panic!`s from the decoder struct and replaced them with instructional messages in order to help users determine why a type is not being indexed. We should probably figure out a nicer way to do this, but that can wait for a later PR.
